sphinx/themes/basic/static/searchtools.js

```diff
@@ -308,8 +308,18 @@ const Search = {
     // search for explicit entries in index directives
     for (const [entry, foundEntries] of Object.entries(indexEntries)) {
       if (entry.includes(queryLower) && (queryLower.length >= entry.length/2)) {
-        for (const [file, id] of foundEntries) {
-          let score = Math.round(100 * queryLower.length / entry.length)
+        for (const [file, id, isMain] of foundEntries) {
+          // Give main entries a high score so they are ordered early
+          // and non-main entries a low score so they are ordered later.
+          //
+          // We only get here if the query length is at least half the entry
+          // length, hence, with a weight of 100, main entries have a minimum
+          // score of 50.  We can choose any weight lower than 50 for non-main
+          // entries to ensure they are always ordered after main entries, but
+          // since they are typically just arbitrary cross-references to the
+          // indexed entity, we choose a very low score.
+          const maxScore = isMain ? 100 : 1;
+          const score = Math.round(maxScore * queryLower.length / entry.length);
```

Comment:
Given the rounding involved, how different is this to setting the score to a fixed value (`0` or `null` for example) for all of the non-index entries?

Maths tricks in scoring can be a useful way to achieve re-ordering, but they don't compose well - so once we begin using one or two of them, it'll become much more difficult to use scoring for any other adjustments.

(my preference would be to avoid the use of mathematical separation of the non-index entries, but to find some other technique, such as appending them separately after the rest of the documents are scored.  it seems debatable though)

Reply:
Oh good catch. I totally missed that point. Indeed, we should perhaps change the maxScore for non-main entries to 49, so that we have a better approximation.

Reply:
I tried 49 locally, but in my use case it still ordered non-main entries before more relevant search results.

I updated this PR to use 10 instead.

> avoid the use of mathematical separation

I've opened #11696 to do that instead.